src/actions/player/playerTrendsActions.js

```diff
@@ -33,33 +36,60 @@ export const getPlayerTrendsError = (payload, id) => ({
   id,
 });
 
+const getCumalativeSums = (matches, fieldName) =>
+  matches.reduce((cumulativeList, match, index) => {
+    if (cumulativeList.length > 0) {
+      const prevTotal = cumulativeList[index - 1];
+      cumulativeList.push(prevTotal + match[fieldName]);
+    } else {
+      cumulativeList.push(match[fieldName]);
+    }
+    return cumulativeList;
+  }, []);
+
+const getWinRateCumlativeSums = matches =>
+  matches.reduce((cumulativeList, match, index) => {
+    // 1 for win 0 for loss
+    const winInt = match.radiant_win === isRadiant(match.player_slot) ? 1 : 0;
+    if (index === 0) {
+      cumulativeList.push(winInt);
+    } else {
+      const prevRate = cumulativeList[index - 1];
+      const newRate = ((prevRate * index) + winInt) / (index + 1);
+      cumulativeList.push(newRate);
+    }
+
+    return cumulativeList;
+  }, []);
+
 export const getPlayerTrends = (playerId, options = {}, fieldName) => (dispatch) => {
   dispatch(getPlayerTrendsRequest(playerId));
   return fetch(`${API_HOST}${getUrl(playerId, options, url)}`)
     .then(response => response.json())
     .then((response) => {
-      const matches = response.filter(match => match[fieldName] !== undefined && match[fieldName] !== null)
-      .reverse();
+      const winRateTrend = fieldName === 'win_rate'; // Special case
+      const matches = winRateTrend
+        ? response.reverse()
+        : response.filter(match => match[fieldName] !== undefined && match[fieldName] !== null)
+          .reverse();
 
-      const cumulativeSums = matches.reduce((cumulativeList, match, index) => {
-        if (cumulativeList.length > 0) {
-          const prevTotal = cumulativeList[index - 1];
-          cumulativeList.push(prevTotal + match[fieldName]);
-        } else {
-          cumulativeList.push(match[fieldName]);
-        }
-        return cumulativeList;
-      }, []);
+      const cumulativeSums = winRateTrend
+        ? getWinRateCumlativeSums(matches)
+        : getCumalativeSums(matches, fieldName);
 
       return cumulativeSums.map((value, index) => {
         const match = matches[index];
         return {
           x: index + 1,
-          value: Number((value / (index + 1)).toFixed(2)),
-          independent_value: match[fieldName],
+          value: winRateTrend
+            ? (value * 100).toFixed(2)
```

Comment:
I think this formatting should be done at the display level.  You can just keep the raw data here?